src/fidesops/api/v1/endpoints/dataset_endpoints.py

```diff
@@ -143,9 +148,19 @@ def patch_datasets(
             "dataset": dataset.dict(),
         }
         try:
+            if connection_config.connection_type == ConnectionType.saas:
+                __validate_saas_dataset(connection_config, dataset)
```

Comment:
Another question about dataset validation for saas connection configs - you can technically add as many datasets as you want to a connection config.  Should there be an additional restriction that you can only add one if it is a saas connection config? Especially since they come in pairs.

Reply:
Enforcing this by checking for matching fides_keys between the dataset and the SaaS config.

There is still the scenario where the user can submit multiple valid datasets with the same fides_key but they would override each other as we iterated through the list. Not sure how to notify the user of this while still conforming to the `BulkPutDataset` response format. Would a `logger.warn` be sufficient?

Reply:
that's a good idea